weather.go

```diff
@@ -1,5 +1,138 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"strings"
+)
+
+type apiResponse struct {
+	Name    string
+	Weather []struct {
+		Main string
+	}
+	Main struct {
+		Temp float64
+	}
+}
+
+type apiErrorResponse struct {
+	Cod     string
+	Message string
+}
+
+type Conditions struct {
+	City               string
+	OneWord            string
+	TemperatureCelsius float64
+}
+
+func RunCLI(args []string) (output string) {
+
+	token := os.Getenv("WEATHERAPP_TOKEN")
+	if len(token) == 0 {
+		fmt.Fprintf(os.Stderr, "please set env variable, WEATHERAPP_TOKEN \n")
+		os.Exit(2)
+	}
+
+	request, err := Request(args, token)
+	if err != nil {
+		log.Printf("an error has occured, %v", err)
+		os.Exit(2)
+	}
+
+	response, err := Response(request)
+	if err != nil {
+		log.Printf("an error has occured, %v", err)
+		os.Exit(2)
+	}
+
+	conditions, err := ParseResponse(response)
+	if err != nil {
+		log.Printf("problem parsing API response', %v", err)
+		os.Exit(2)
+	}
+
+	output = fmt.Sprintf("City: %s\nWeather: %s\nCelsius: %v\n", conditions.City, conditions.OneWord, conditions.TemperatureCelsius)
+
+	return output
+}
+
+func Request(args []string, token string) (*http.Request, error) {
+
+	domain := "api.openweathermap.org"
+
+	if len(os.Args) == 1 {
```

Comment:
```suggestion
	if len(os.Args) < 2 {
```

On its own, `== 1` conveys no information about what you _need_. But `< 2` tells the reader you _need two arguments_.